### ISSUE TITLE
When offline cache is not Optimistic Updated

### ISSUE THREAD

**Issue by KyleBrost** [aerogear/offix#433]
I have a mutation:

```
    const theOptimisticResponse = createOptimisticResponse({
      mutation: EMPLOYEE_TIMESHEET_ENTRY_MUTATE_UPDATE,
      variables: {
        ete_id : updateData_.ete_id,
        ete_comment: updateData_.ete_comment,
        ete_start_time: updateData_.ete_start_time,
        ete_stop_time: updateData_.ete_stop_time,
        employee_id: updateData_.employee_id,
      },
      operationType: CacheOperation.ADD,
      returnType: 'EmployeeTimesheetEntry',
      idField: 'ete_id'
    });

    this.client.offlineMutate({mutation: EMPLOYEE_TIMESHEET_ENTRY_MUTATE_UPDATE,
                               variables: {updateData: updateData_},
                               idField: "ete_id",
                               returnType: "EmployeeTimesheetEntry",
                               optimisticResponse:theOptimisticResponse,
                               updateQuery: {
                                query: EMPLOYEE_TIMESHEET_ENTRY_QUERY_ALL,
                                  variables: { startDate: theDate }
                               },
                              })
    .then( (result) => {
      console.log("### MUTATE COMPLETE");
      console.dir(result);
    })
    .catch( (error) => {
      console.log("### MUTATE ERROR");
      console.error(error);
    })

```

When things are online everything works fine.  When the client is offline everything does resolve and update when it goes back online.  But what is not happening for ADD operations:

1. Load view and see latest data
2. Go offline
3. Add a new record
4. View does not update
5. Go online
6. View update and is correct

If we do the same for an "update" operation the view is updated.

What am I missing to get newly created items to appear optimiticly when offline?

**Comment by darahayes** [comment]
Thanks for submitting this issue @KyleBrost. On the face of it, what you are doing looks correct and *should* work especially if you have no issues with updating existing items. Sorry you're not having the best experience.

I suspect that this issue might be related to #430. In that issue, another user had problems with the `createOptimisticResponse` function because they were passing their own ID value as part of the create mutation. Based on the example code you showed above, you are also passing your own ID value when attempting to add a new record. It's possible that the bug discovered in the issue I linked is causing the view to not update. I will investigate and get back to you as soon as I have more information. In the meantime, you can try out the workaround suggested in that issue and see if it makes a difference for you.

**Comment by darahayes** [comment]
@KyleBrost we've just published version `0.15.1` of `offix-client` can you please try it out and let me know if you still experience this issue? Thank you!

**Comment by wtrocki** [comment]
Thank you for reporting issue. Feel free to reopen if the issue still persist
